apps/settings/src/components/Users/UserRow.vue

```diff
@@ -425,13 +425,27 @@ export default {
 
 		userGroupsLabels() {
 			return this.userGroups
-				.map(group => group.name ?? group.id)
+				.map(group => {
+					// Try to match with more extensive group data
+					const availableGroup = this.availableGroups.find(g => g.id === group.id)
+					if (availableGroup) {
+						return availableGroup.name ?? availableGroup.id
+					}
+					return group.name ?? group.id
```

Comment:
Also works in 1 line if preferred:
```suggestion
				return availableGroup.name ?? group.name ?? group.id
```